lib/spack/spack/test/data/modules/tcl/suffix.yaml

```diff
@@ -9,3 +9,4 @@ tcl:
       '+debug': foo
       '^mpich': foo
       '~debug': bar
+      '^mpi': 'mpi={^mpi.name}-v{^mpi.version}'
```

Comment:
Can we write unit-tests specific for this feature, instead of modifying existing ones? Can we also test cases that were previously not supported?

Reply:
Done.

> Can we also test cases that were previously not supported?

Not sure if you meant something else?